src/Admin/templates/globals/globals.html.twig

```diff
@@ -0,0 +1,28 @@
+<!doctype html>
+<html lang="en">
+<head>
+    {% include "@admin/partials/head.html.twig" %}
+</head>
+<body class="body_top">
+{% include "@admin/partials/contextbar.html.twig" %}
+
+<div class="container-fluid">
+    <div class="row">
+        {% include "@admin/partials/sidebar.html.twig" %}
+        <div class="col-xs-12 col-sm-9 col-sm-offset-2 main-area">
+            <form method='post' name='theform' id='theform' class='form-horizontal' action="{{ form_action }}">
+                <div class="tab-content">
+                    <input type="hidden" name="csrf_token_form" value="{{ csrf|e('html_attr') }}" />
```

Comment:
Would it make sense to create our own escaping functions?(would basically just wrap the ones we already use throughout the codebase). I simply don't trust generic ones and then easy to globally adjust if they are ours.

Reply:
And we could also make wrappers for the translation/escaping combo functions.

Reply:
Yes, we _could_ create additional functions. The `translate` filter is already doing just what you suggest. However, look at the underlying escape function in Twig and in our own custom function... it's the same thing. And I'd trust a templating engine's escape functions over our own.

Reply:
Hi @robertdown ,

Escaping for simple stuff is much more complicated than escaping for OpenEMR :)

The current escaping functions are here:
https://github.com/openemr/openemr/blob/master/library/htmlspecialchars.inc.php

Note there are 6 escaping functions and 4 additional escaping functions for translations. And for the record, I do not trust a templating engine's escape functions over our own :)

Reply:
Our escaping functions are simple wrappers, mostly for either json_encoding or htmlspecialchars... what's Twig use for their escaping? htmlspecialchars.

https://github.com/twigphp/Twig/blob/2.x/src/Extension/EscaperExtension.php

Escaping AND translating together would still require our internal functions, but remember your earlier position that we shouldn't be escaping variables in the templates because it'll break the pipeline.